packages/main/src/Switch.ts

```diff
@@ -212,22 +212,26 @@ class Switch extends UI5Element {
 		}
 
 		if (isEnter(e)) {
-			this.toggle();
+			this._onclick();
 		}
 	}
 
 	_onkeyup(e: KeyboardEvent) {
 		if (isSpace(e)) {
-			this.toggle();
+			this._onclick();
 		}
 	}
 
 	toggle() {
 		if (!this.disabled) {
-			this.checked = !this.checked;
-			this.fireEvent("change");
+			const changePrevented = !this.fireEvent("change", null, true);
 			// Angular two way data binding;
-			this.fireEvent("value-changed");
+			const valueChagnePrevented = !this.fireEvent("value-changed", null, true);
+			this.checked = !this.checked;
```

Comment:
this needs to be before the event is fired (like it was before), otherwise the event handler will still have the previous value